orbit/cmd/orbit/orbit.go

```diff
@@ -1244,15 +1246,34 @@ func (d *desktopRunner) execute() error {
 		opts = append(opts, execuser.WithEnv("FLEET_DESKTOP_INSECURE", "1"))
 	}
 
+	var buf bytes.Buffer
+
 	for {
 		// First retry logic to start fleet-desktop.
 		if done := retry(30*time.Second, false, d.interruptCh, func() bool {
+			wcmd := exec.Command("w")
+			wcmd.Stderr = os.Stderr
+			wcmd.Stdout = &buf
+			if err := wcmd.Run(); err != nil {
+				log.Error().Err(err).Msg("calling w")
+			}
+
+			if !strings.Contains(buf.String(), "console") {
+				log.Info().Msg("no user found logged into GUI")
+				return true
+			}
+
+			buf.Reset()
```

Comment:
derp I should reset no matter what I return. also, I think there are race conditions that can affect this buffer and lead to the resets not working. I think I need to add some locks to avoid that. I'll update this.